From 5287d03b6fa87963c433e2f0feefeaced8dc9f94 Mon Sep 17 00:00:00 2001
From: Konrad Weihmann <kweihmann@outlook.com>
Date: Wed, 16 Dec 2020 21:14:51 +0100
Subject: [PATCH] Makefile: cross compile fixes

Signed-off-by: Konrad Weihmann <kweihmann@outlook.com>
Upstream-Status: Inappropriate [oe-specific]
---
 template/Makefile.in | 15 ++++++++-------
 1 file changed, 8 insertions(+), 7 deletions(-)

diff --git a/template/Makefile.in b/template/Makefile.in
index d2df569..3ad6896 100644
--- a/template/Makefile.in
+++ b/template/Makefile.in
@@ -324,19 +324,20 @@ rbconfig.rb: $(RBCONFIG)
 install-cross: $(arch)-fake.rb $(RBCONFIG) rbconfig.rb $(arch_hdrdir)/ruby/config.h \
 	$(LIBRUBY_A) $(LIBRUBY_SO) $(ARCHFILE)
 	$(ECHO) installing cross-compiling stuff
-	$(Q) $(MAKEDIRS) $(XRUBY_RUBYLIBDIR)/$(arch) $(XRUBY_RUBYHDRDIR)/$(arch)/ruby
+	$(Q) $(MAKEDIRS) $(DESTDIR)$(XRUBY_RUBYLIBDIR)/$(arch)-cross $(DESTDIR)$(XRUBY_RUBYHDRDIR)/$(arch)-cross/ruby
 	$(Q) sed '/^\$$:\.unshift/q' $(arch)-fake.rb > fake.rb
 	$(Q) $(BASERUBY) -p \
 	-e '~/^\s*CONFIG\["LDFLAGS"\]/ and' \
 	-e '$$_[/(?=\s*"$$)/] = %q[ #{(CONFIG["LIBPATHFLAG"]%File.dirname(__FILE__)).strip}]' \
 	rbconfig.rb > fake-rbconfig.rb
-	$(INSTALL_SCRIPT) fake.rb $(XRUBY_RUBYLIBDIR)/$(arch)/fake.rb
-	$(INSTALL_SCRIPT) fake-rbconfig.rb $(XRUBY_RUBYLIBDIR)/$(arch)/rbconfig.rb
+
+	$(INSTALL_SCRIPT) fake.rb $(DESTDIR)$(XRUBY_RUBYLIBDIR)/$(arch)-cross/fake.rb
+	$(INSTALL_SCRIPT) fake-rbconfig.rb $(DESTDIR)$(XRUBY_RUBYLIBDIR)/$(arch)-cross/rbconfig.rb
 	@$(RM) fake.rb fake-rbconfig.rb
-	$(INSTALL_DATA) $(arch_hdrdir)/ruby/config.h $(XRUBY_RUBYHDRDIR)/$(arch)/ruby
-	$(INSTALL_DATA) $(top_srcdir)/include/ruby/win32.h $(XRUBY_RUBYHDRDIR)/ruby
-	$(INSTALL_DATA) $(LIBRUBY) $(LIBRUBY_A) $(XRUBY_RUBYLIBDIR)/$(arch)
-	$(INSTALL_PROGRAM) $(LIBRUBY_SO) $(XRUBY_RUBYLIBDIR)/$(arch)
+	$(INSTALL_DATA) $(arch_hdrdir)/ruby/config.h $(DESTDIR)$(XRUBY_RUBYHDRDIR)/$(arch)-cross/ruby
+	$(INSTALL_DATA) $(LIBRUBY) $(DESTDIR)$(XRUBY_RUBYLIBDIR)/$(arch)-cross
+	$(INSTALL_DATA) $(LIBRUBY_A) $(DESTDIR)$(XRUBY_RUBYLIBDIR)/$(arch)-cross
+	$(INSTALL_PROGRAM) $(LIBRUBY_SO) $(DESTDIR)$(XRUBY_RUBYLIBDIR)/$(arch)-cross
 
 Makefile:	$(srcdir)/template/Makefile.in $(srcdir)/enc/Makefile.in
 
-- 
2.25.1

